ui/app/AppLayouts/Wallet/services/dapps/DappsConnectorSDK.qml

```diff
@@ -0,0 +1,133 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import QtWebEngine 1.10
+import QtWebChannel 1.15
+
+import StatusQ.Core.Utils 0.1 as SQUtils
+import StatusQ.Components 0.1
+
+import StatusQ 0.1
+import SortFilterProxyModel 0.2
+import AppLayouts.Wallet.controls 1.0
+import shared.popups.walletconnect 1.0
+import AppLayouts.Wallet.services.dapps 1.0
+import AppLayouts.Wallet.services.dapps.types 1.0
+
+import shared.stores 1.0
+import utils 1.0
+
+import "types"
+
+// Act as another layer of abstraction to the WalletConnectSDKBase
+// Quick hack until the WalletConnectSDKBase could be refactored to a more generic DappProviderBase with API to match
+// the UX requirements
+WalletConnectSDKBase {
+    id: root
+
+    // Nim connector.controller instance
+    property var controller
+    property bool sdkReady: true
+    property bool active: true
+    required property WalletConnectService wcService
+    property string requestID: ""
+
+    projectId: ""
+
+    implicitWidth: 1
+    implicitHeight: 1
+
+    Loader {
+        id: connectDappLoader
+
+        active: false
+
+        property var dappChains: []
+        property var sessionProposal: null
+        property var availableNamespaces: null
+        property var sessionTopic: null
+        readonly property var proposalMedatada: !!sessionProposal
+                                                ? sessionProposal.params.proposer.metadata
+                                                : { name: "", url: "", icons: [] }
+
+        sourceComponent: ConnectDAppModal {
+            visible: true
+
+            onClosed: connectDappLoader.active = false
+            accounts: root.wcService.validAccounts
+            property SortFilterProxyModel filteredFlatModel: SortFilterProxyModel {
```

Comment:
let's add a sortRole so Ethereum main net goes first.

Reply:
Ethereum mainnet is first in the list already. I have updated the `NetworkSelectorView`. I am not sure why the decision was to select the last element. But the change shouldn't be a big deal.